src/@hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,42 @@
+import { useState } from 'react'
+
+function useLocalStorage<T>(key: string, initialValue: T) {
+  // State to store our value
+  // Pass initial state function to useState so logic is only executed once
+  const [storedValue, setStoredValue] = useState<T>(() => {
+    if (typeof window === 'undefined') {
+      return initialValue
+    }
+    try {
+      // Get from local storage by key
+      const item = window.localStorage.getItem(key)
+      // Parse stored json or if none return initialValue
+      return item ? JSON.parse(item) : initialValue
+    } catch (error) {
+      // If error also return initialValue
+      console.log(error)
+      return initialValue
+    }
+  })
+  // Return a wrapped version of useState's setter function that ...
+  // ... persists the new value to localStorage.
+  const setValue = (value: T | ((val: T) => T)) => {
```

Comment:
not sure if we can reuse part of the work or move some into the  UserPreferencesProvider we have under @context/UserPreference.tsx

Reply:
I don't quite understand what you mean here? Should I move the hook inside UserPreferencesProvider? But then again the stored data has nothing to do with user preferences.

Reply:
Yep, makes sense, I think this can stay as it is